### ISSUE TITLE
undefined method `supported_ssl_methods'

### ISSUE THREAD

**Issue by esfomeado** [rapid7/metasploit-framework#9800]
## Steps to reproduce

How'd you do it?

1. Execute msfvenom -p windows/meterpreter/reverse_tcp --payload-options

## Expected behavior

Should show the payload options

## Current behavior

```
Traceback (most recent call last):
	37: from /usr/bin/msfvenom:321:in `<main>'
	36: from /usr/bin/msfvenom:55:in `framework'
	35: from /usr/bin/msfvenom:46:in `init_framework'
	34: from /usr/share/metasploit-framework/lib/msf/base/simple/framework.rb:73:in `create'
	33: from /usr/share/metasploit-framework/lib/msf/base/simple/framework.rb:121:in `simplify'
	32: from /usr/share/metasploit-framework/lib/msf/base/simple/framework/module_paths.rb:49:in `init_module_paths'
	31: from /usr/share/metasploit-framework/lib/msf/base/simple/framework/module_paths.rb:49:in `each'
	30: from /usr/share/metasploit-framework/lib/msf/base/simple/framework/module_paths.rb:50:in `block in init_module_paths'
	29: from /usr/share/metasploit-framework/lib/msf/core/module_manager/module_paths.rb:40:in `add_module_path'
	28: from /usr/share/metasploit-framework/lib/msf/core/module_manager/module_paths.rb:40:in `each'
	27: from /usr/share/metasploit-framework/lib/msf/core/module_manager/module_paths.rb:41:in `block in add_module_path'
	26: from /usr/share/metasploit-framework/lib/msf/core/module_manager/loading.rb:117:in `load_modules'
	25: from /usr/share/metasploit-framework/lib/msf/core/module_manager/loading.rb:117:in `each'
	24: from /usr/share/metasploit-framework/lib/msf/core/module_manager/loading.rb:119:in `block in load_modules'
	23: from /usr/share/metasploit-framework/lib/msf/core/modules/loader/base.rb:237:in `load_modules'
	22: from /usr/share/metasploit-framework/lib/msf/core/modules/loader/directory.rb:30:in `each_module_reference_name'
	21: from /usr/share/metasploit-framework/lib/msf/core/modules/loader/directory.rb:30:in `foreach'
	20: from /usr/share/metasploit-framework/lib/msf/core/modules/loader/directory.rb:40:in `block in each_module_reference_name'
	19: from /usr/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/rex-core-0.1.13/lib/rex/file.rb:132:in `find'
	18: from /usr/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/rex-core-0.1.13/lib/rex/file.rb:132:in `catch'
	17: from /usr/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/rex-core-0.1.13/lib/rex/file.rb:133:in `block in find'
	16: from /usr/share/metasploit-framework/lib/msf/core/modules/loader/directory.rb:50:in `block (2 levels) in each_module_reference_name'
	15: from /usr/share/metasploit-framework/lib/msf/core/modules/loader/base.rb:238:in `block in load_modules'
	14: from /usr/share/metasploit-framework/lib/msf/core/modules/loader/base.rb:183:in `load_module'
	13: from /usr/share/metasploit-framework/lib/msf/core/module_manager/loading.rb:73:in `on_module_load'
	12: from /usr/share/metasploit-framework/lib/msf/core/payload_set.rb:198:in `add_module'
	11: from /usr/share/metasploit-framework/lib/msf/core/payload_set.rb:198:in `new'
	10: from /usr/share/metasploit-framework/modules/payloads/stages/python/meterpreter.rb:25:in `initialize'
	 9: from /usr/share/metasploit-framework/lib/msf/core/payload/python/meterpreter_loader.rb:45:in `initialize'
	 8: from /usr/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/activesupport-4.2.10/lib/active_support/dependencies.rb:274:in `require'
	 7: from /usr/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/activesupport-4.2.10/lib/active_support/dependencies.rb:240:in `load_dependency'
	 6: from /usr/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/activesupport-4.2.10/lib/active_support/dependencies.rb:274:in `block in require'
	 5: from /usr/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/backports-3.11.1/lib/backports/std_lib.rb:9:in `require_with_backports'
	 4: from /usr/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/backports-3.11.1/lib/backports/std_lib.rb:9:in `require'
	 3: from /usr/share/metasploit-framework/lib/msf/core/opt.rb:3:in `<top (required)>'
	 2: from /usr/share/metasploit-framework/lib/msf/core/opt.rb:16:in `<module:Msf>'
	 1: from /usr/share/metasploit-framework/lib/msf/core/opt.rb:111:in `<module:Opt>'
/usr/share/metasploit-framework/lib/msf/core/opt.rb:57:in `SSLVersion': undefined method `supported_ssl_methods' for Rex::Socket::SslTcp:Module (NoMethodError)
```

You might also want to check the last ~1k lines of
`/opt/metasploit/apps/pro/engine/config/logs/framework.log` or
`~/.msf4/logs/framework.log` for relevant stack traces

### Metasploit version

Framework: 4.16.47-dev
Console  : 4.16.47-dev

### I installed Metasploit with:
- [x] Kali package via apt
- [ ] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [ ] Source install (please specify ruby version)

### OS

What OS are you running Metasploit on?
 4.12.0-kali2-686

**Comment by busterb** [comment]
Hmm, @wvu-r7 is this related to this upstream Kali patch? http://git.kali.org/gitweb/?p=packages/metasploit-framework.git;a=blob;f=debian/patches/add-missing-requirements-msfvenom.patch;h=05a213e75afdaee549516b03c07fd2ecc6515abb;hb=d07ad0906a20d5c2334a0eea50e2c75fc9f38503

Do we need that too?

**Comment by ChopinDebussy** [comment]
Hi, 
Try to add:
`require '/use/share/metasploit-framework/vendor/bundle/ruby/2.5.0/gems/rex-socket-0.1.13/lib/rex/socket/ssl_tcp.rb' `
to opt.rb file (/usr/share/metasploit-framework/lib/msf/core)
It solved my problem.

**Comment by busterb** [comment]
Thanks for the note @ChopinDebussy

**Comment by busterb** [comment]
Running `bundle exec ./msfvenom -p windows/meterpreter/reverse_tcp --payload-options` also resolves it.  Hmm...

**Comment by busterb** [comment]
running `bundle exec` just as in #9800 fixes this as well.

**Comment by jmartin-r7** [comment]
Fixed with #9817, this will take a day or so to filter down to apt packages.
